### ISSUE TITLE
Clean up all nodes in prep for pub-sub

### ISSUE THREAD

**Issue by kylestach** [RoboJackets/robocup-software#1424]
## Description
As preparation for the ROS port, we want to clean up all of our code to make it more conducive to a pub-sub framework. This means making all of our nodes do something like the following:
```cpp
class SomeNode : public Node {
public:
  void run() override {
    runNode(_context->world_state, &_context->robot_intents);
  }

private:
  Context* _context;
  void runNode(const WorldState& state, RobotIntent* intent);
};
```
Basically, `runNode` (or whatever it is called) should only take in the parts of Context that it actually reads/writes. Nothing else from Context should be directly accessed with the exception of debug-drawing facilities. This enforces that the node's application logic only uses parts of Context that it has declared as dependencies.

## Requirements

The following nodes should not use Context other than in their top-level interface:
 - [x] Motion control (@kylestach)
 - [x] GUI (@kylestach @chachmu)
 - [x] Radio (@HussainGynai)
 - [x] Vision receiver
 - [x] Vision filter (pretty much already done)
 - [x] Manual control (@CollinAvidano)
 - [x] Referee (@oswinso)
 - [x] GrSim (this is already fine)
 - [x] Planning (@egordon9dev, waiting on refactor to be complete)
For now, gameplay will not be changed (the interface to context will need to be rewritten anyway).

Deadline targets are listed in the below document.

## Design

[Roadmap meeting](https://docs.google.com/document/d/1wix-ZK9NPBY0zC8T9LyUihT1ONubXMKxHvb8KYmJkNQ/edit#)
